### PR TITLE
Fixes for internal runtime usage in build and PRs

### PR DESCRIPTION
**Pull request by mmitche**
- Don't conditionalize queue adds based on use of open queues.
  In eng/targets/Helix.targets, these are replaced with the non-open versions automatically
- Don't use default error detection for the repo tasks build in source build, as this may
  pick up the error messages from the first attempts to install an internal runtime from a public location (it later succeeds using the private location.)

### CODE REVIEW

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -32,21 +32,21 @@
...
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true' AND '$(_UseHelixOpenQueues)' == 'true'">
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true'">
     <!-- TODO Re-enable Win-7 queue when dotnet restore are fixed. https://github.com/dotnet/aspnetcore/issues/32683 -->
     <!-- <HelixAvailableTargetQueue Include="Windows.7.Amd64.Open" Platform="Windows" /> -->
     <HelixAvailableTargetQueue Include="Windows.81.Amd64.Open" Platform="Windows" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true' AND '$(_UseHelixOpenQueues)' != 'true'">
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' == 'true'">
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.ClientPre.VS2019.Pre" Platform="Windows" />
```
Comment (Tratcher) [inline eng/targets/Helix.Common.props:41]: This queue doesn't have an Open version does it? Can the automatic logic handle that?

Comment (mmitche) [reply eng/targets/Helix.Common.props:41]: Good point. Let me reintroduce that.

**Comment by Tratcher** [comment]
What do we expect to happen to this build queue?
https://dev.azure.com/dnceng/internal/_build?definitionId=928

It's intent is to only run `Windows.10.Amd64.ClientPre.VS2019.Pre`. Will it start running the non-open versions of all queues now?

**Comment by mmitche** [comment]
> What do we expect to happen to this build queue?
> https://dev.azure.com/dnceng/internal/_build?definitionId=928
> 
> It's intent is to only run `Windows.10.Amd64.ClientPre.VS2019.Pre`. Will it start running the non-open versions of all queues now?

There is replacement logic in the repo that will replace '.Open' with the empty string when running on the internal project, where the .Open queues are not available. In public, the .Open queues are only available, and nothing changes.

**Comment by Tratcher** [comment]
We don't want all of those jobs running on that internal queue, there's a public queue that runs them separately.
https://dev.azure.com/dnceng/public/_build?definitionId=837

Is there another internal queue where you're trying to enable these jobs? Do we need a different mechanic to disable these for the existing internal queue?

**Comment by mmitche** [comment]
> We don't want all of those jobs running on that internal queue, there's a public queue that runs them separately.
> https://dev.azure.com/dnceng/public/_build?definitionId=837
> 
> Is there another internal queue where you're trying to enable these jobs? Do we need a different mechanic to disable these for the existing internal queue?

This is for internal PRs only, not for enabling on official builds or whatnot. Right now no Helix tests are running on internal PRs. The intention is to mirror the public PRs in this case.

**Comment by Tratcher** [comment]
But as a side-effect, they'll all start running on https://dev.azure.com/dnceng/internal/_build?definitionId=928, right? We'd want a way to suppress that.

**Comment by mmitche** [comment]
> But as a side-effect, they'll all start running on https://dev.azure.com/dnceng/internal/_build?definitionId=928, right? We'd want a way to suppress that.

Good point. I'll look into getting that fixed.

**Review by HaoK — APPROVED** [review]

**Comment by mmitche** [comment]
@Tratcher Fixed this up. The previous logic on queue additions is a bit hard to follow (since the ci.yml queues ended up getting used for the helix-matrix and quarantine cases). I fixed it up by making the first block specific to PRs only and duplicating those 3 queues into the daily blocks in the right places.

If this queue matrix ever gets more complicated it probably deserves a rework so that it's not a mishmash of conditionals. In these cases I tend to favor using some kind of enumeration identifier for each distinct scenario and then using the `Choose` construct, making each scenario have an explicit queue list. It does result in duplication of lists, but it's a lot easier to follow and results in less unintended side effects when someone who doesn't understand subtleties in the conditionals attempts to change something.

**Comment by Tratcher** [comment]
Thanks @mmitche, sounds good to me. I've pinged Hao for a fresh review of the implementation.

File: eng/SourceBuild.props [hunk 2]
```diff
@@ -46,8 +46,20 @@
           Condition="'$(ArcadeInnerBuildFromSource)' == 'true'"
           BeforeTargets="Execute">
 
+    <!-- If the alternative runtime location and key are present, pass those through -->
+    <PropertyGroup>
+      <_AdditionalRepoTaskBuildArgs></_AdditionalRepoTaskBuildArgs>
```
Comment (dougbu) [inline eng/SourceBuild.props:51]: nit
```suggestion
      <_AdditionalRepoTaskBuildArgs />
```

File: eng/targets/Helix.Common.props [hunk 3]
```diff
@@ -10,41 +10,55 @@
...
+  <ItemGroup Condition="'$(IsRequiredCheck)' == 'true' AND '$(TargetArchitecture)' == 'x64' AND '$(IsHelixDaily)' != 'true'">
     <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.Server20H2.Open" Platform="Windows" />
     <HelixAvailableTargetQueue Include="OSX.1014.Amd64.Open" Platform="OSX" />
   </ItemGroup>
   
-  <!-- Quarantined test queues -->
+  <!-- x64 Quarantined-only (quarantined-pr.yml and quarantined-tests.yml) test queues -->
   <ItemGroup Condition="'$(IsRequiredCheck)' == 'true' AND '$(TargetArchitecture)' == 'x64' AND '$(_UseHelixOpenQueues)' == 'true' AND '$(RunQuarantinedTests)' == 'true'">
     <HelixAvailableTargetQueue Include="(Fedora.34.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125" Platform="Linux" />
```
Comment (dougbu) [inline eng/targets/Helix.Common.props:22]: Can we please change the base image while we're here❔
```suggestion
    <HelixAvailableTargetQueue Include="(Fedora.34.Amd64.Open)ubuntu.1804.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125" Platform="Linux" />
```

/fyi @wtgodbe

File: eng/targets/Helix.Common.props [hunk 4]
```diff
@@ -10,41 +10,55 @@
...
     <HelixAvailableTargetQueue Include="Redhat.7.Amd64.Open" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="Ubuntu.2004.Amd64.Open" Platform="Linux" />
+    <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
+
+    <!-- Mac -->
+    <HelixAvailableTargetQueue Include="OSX.1014.Amd64.Open" Platform="OSX" />
+    <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="OSX" />
+
+    <!-- Containers -->
     <HelixAvailableTargetQueue Include="(Fedora.34.Amd64.Open)ubuntu.1604.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125" Platform="Linux" />
```
Comment (dougbu) [inline eng/targets/Helix.Common.props:38]: here too
```suggestion
    <HelixAvailableTargetQueue Include="(Fedora.34.Amd64.Open)ubuntu.1804.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-34-helix-20210728124700-4f64125" Platform="Linux" />
```

**Review by HaoK — APPROVED** [review]
Sounds reasonable to me, eventually switching to explicit lists for each pipeline/usage does seem inevitable at this point
